### PR TITLE
Refactor image tests to gomega

### PR DESCRIPTION
**Pull request by adambkaplan**
Replace `t.Fatal` and `t.Fatalf` with gomega equivalents

### CODE REVIEW

**Comment by deads2k** [comment]
@adambkaplan @dmage this is important for debugging jobs, can you guys find review time to get it in?

File: test/extended/images/imagestream.go [hunk 1]
```diff
@@ -258,59 +232,38 @@ func TestImageStreamWithoutDockerImageConfig(t g.GinkgoTInterface, oc *exutil.CL
...
-	}
-	if ist.Image.Name != image.Name {
-		t.Fatalf("unexpected image: %#v", img)
-	}
-	if len(ist.Image.DockerImageConfig) != 0 {
-		t.Errorf("image has a not empty config: %#v", ist)
-	}
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(ist.Image.Name).To(o.Equal(image.Name))
+	o.Expect(ist.Image.DockerImageConfig).NotTo(o.BeEmpty())
```
Comment (dmage) [inline test/extended/images/imagestream.go]: it should be empty

**Comment by dmage** [comment]
/hold
one test should be fixed to be green

**Comment by adambkaplan** [comment]
@dmage fixed the broken test

File: test/extended/images/imagestream.go [hunk 2]
```diff
@@ -258,59 +232,38 @@ func TestImageStreamWithoutDockerImageConfig(t g.GinkgoTInterface, oc *exutil.CL
...
-	if isi.Image.Name != image.Name {
-		t.Fatalf("unexpected image: %#v", img)
-	}
-	if len(isi.Image.DockerImageConfig) != 0 {
-		t.Errorf("image has a not empty config: %#v", isi)
-	}
-
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(isi.Image.Name).To(o.Equal(image.Name))
+	o.Expect(isi.Image.DockerImageConfig).NotTo(o.BeEmpty())
```
Comment (dmage) [inline test/extended/images/imagestream.go]:
```
fail [github.com/openshift/origin/test/extended/images/imagestream.go:251]: Expected
    <string>: 
not to be empty
```

**Pull request merged**